File: src/main/java/ro/nicolaemariusghergu/queryit/service/ManufacturerService.java
package ro.nicolaemariusghergu.queryit.service;

import org.springframework.lang.NonNull;
import ro.nicolaemariusghergu.queryit.model.Manufacturer;

import java.util.List;
import java.util.Optional;

public interface ManufacturerService {

    @NonNull
    Optional<Manufacturer> findById(@NonNull Long id);

    @NonNull
    List<Manufacturer> findAll();

    Optional<Manufacturer> findByName(String name);
}

File: src/main/kotlin/ro/nicolaemariusghergu/queryit/service/ManufacturerService.kt
package ro.nicolaemariusghergu.queryit.service

import org.springframework.http.ResponseEntity
import ro.nicolaemariusghergu.queryit.dto.ManufacturerDto


interface ManufacturerService {
    fun findManufacturerById(id: Long): ResponseEntity<ManufacturerDto>

    fun getManufacturers(): ResponseEntity<List<ManufacturerDto>>

    fun addManufacturer(manufacturerDto: ManufacturerDto): ResponseEntity<Long>

    fun getManufacturerByName(name: String): ResponseEntity<ManufacturerDto>

    fun deleteManufacturerById(id: Long): ResponseEntity<Long>

    fun updateManufacturer(manufacturerDto: ManufacturerDto): ResponseEntity<ManufacturerDto>
}